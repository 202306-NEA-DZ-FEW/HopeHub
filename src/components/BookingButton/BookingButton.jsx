import Link from "next/link";
import { useTranslation } from "next-i18next";

export default function BookingButton() {
    //Function used for translations
    const { t } = useTranslation("common");

    //Rendering the button that leads to booking
    return (
        <div>
            <button
                className='button-container bg-Accent dark:bg-Dark_Accent rounded-md w-36 h-7 sm:w-52 sm:h-12 lg:w-64 lg:h-14 hover:bg-NeutralWhite dark:hover:bg-NeutralBlack'
                data-testid='booking-button'
            >
                <Link
                    href='/booking'
                    className='text-xs sm:text-base lg:text-xl text-NeutralBlack dark:text-NeutralWhite uppercase font-normal font-poppins'
                >
                    {t("Book An Appointement")}
                </Link>
            </button>
        </div>
    );
}
