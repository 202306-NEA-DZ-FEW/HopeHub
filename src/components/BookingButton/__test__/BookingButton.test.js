import renderer from "react-test-renderer";
import BookingButton from "../BookingButton";

it("renders correctly", () => {
    const tree = renderer
        .create(
            <BookingButton
                destination='/booking'
                buttonText='Book An Appointment'
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
