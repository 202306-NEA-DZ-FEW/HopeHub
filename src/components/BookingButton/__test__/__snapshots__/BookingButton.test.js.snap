// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    className="w-72 h-10 rounded-md text-xs sm:text-base lg:text-xl font-poppins font-regular bg-Accent text-NeutralBlack dark:text-NeutralWhite uppercase   dark:bg-Dark_Accent dark:hover:bg-[#3E4E68]  hover:bg-[#879AB8] hover:text-NeutralWhite hover:scale-105 duration-500"
    data-testid="booking-button"
  >
    <a
      href="/booking"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Book An Appointment
    </a>
  </button>
</div>
`;
